src/panels/config/helpers/forms/ha-counter-form.ts

```diff
@@ -155,14 +173,18 @@ class HaCounterForm extends LitElement {
     ev.stopPropagation();
     const target = ev.target as any;
     const configValue = target.configValue;
-    const value =
-      target.type === "number"
-        ? target.value !== ""
-          ? Number(target.value)
-          : undefined
-        : target.localName === "ha-switch"
-          ? (ev.target as HaSwitch).checked
-          : ev.detail?.value || target.value;
+    let value = ev.detail?.value ?? target.value;
+
+    if (target.type === "number") {
+      if (target.value !== "") {
+        value = Number(target.value);
```

Comment:
```suggestion
      if (value !== "") {
        value = Number(value);
```

Reply:
Can you provide some more details for this change?

It could be possible that I'm missing something, but I've only converted the ternary operators to regular if statements to improve the readability.

Reply:
You put `target.value` in the variable `value` above, to avoid confusion/issues in the future, you should then only use that new variable.

Reply:
I see what you mean, it has been implemented :+1: